docs/src/Fieldset.doc.js

```diff
@@ -0,0 +1,222 @@
+// @flow strict
+import type { Node } from 'react';
+import { Fieldset } from 'gestalt';
+import PropTable from './components/PropTable.js';
+import PageHeader from './components/PageHeader.js';
+import MainSection from './components/MainSection.js';
+
+const cards: Array<Node> = [];
+const card = (c) => cards.push(c);
+
+card(
+  <PageHeader
+    name="Fieldset"
+    description="Creates an accessible fieldset and legend for a group of related form items"
+    defaultCode={`
+      function RadioButtonExample() {
+        const [favorite, setFavorite] = React.useState(undefined);
+
+        return (
+          <Fieldset legend="What is your favorite pet?">
+            <Flex direction="column" gap={2}>
+              <RadioButton
+                checked={favorite === 'dogs'}
+                id="favoriteDog"
+                label="Dogs"
+                name="favorite"
+                onChange={() => setFavorite( 'dogs' )}
+                value="dogs"
+              />
+              <RadioButton
+                checked={favorite === 'cats'}
+                id="favoriteCat"
+                label="Cats"
+                name="favorite"
+                onChange={() => setFavorite( 'cats' )}
+                value="cats"
+              />
+              <RadioButton
+                checked={favorite === 'plants'}
+                id="favoritePlants"
+                label="Plants"
+                name="favorite"
+                onChange={() => setFavorite( 'plants' )}
+                value="plants"
+              />
+            </Flex>
+          </Fieldset>
+
+        );
+      }
+`}
+  />,
+);
+
+card(
+  <PropTable
+    Component={Fieldset}
+    props={[
+      {
+        name: 'legend',
+        type: 'string',
+        required: true,
+        description:
+          "Content for this fieldset's legend that clearly and concisely describes the question being asked.",
```

Comment:
Not strong about it, googled Fieldset here and there and they use the word caption. \

Maybe something like 
- "Caption for the form elements grouped in the fieldset" 

I guess I'm iffy with this part `describes the question being asked`. It can be a simple color or gender.